presto-main/src/main/resources/webapp/assets/query-list.js

```diff
@@ -0,0 +1,509 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ var Query = React.createClass({
+    render: function()
+    {
+        var query = this.props.query;
+        var summary = query.state;
+        var progress = 0;
+
+        var completedDrivers = query.completedDrivers;
+        var runningDrivers = query.runningDrivers;
+        var queuedDrivers = query.queuedDrivers;
+
+        // construct query summary and compute progress
+        switch (query.state) {
+            case "FAILED":
+                progress = 100;
+                summary = getReadableErrorCode(query.errorType, query.errorCode);
+                runningDrivers = 0;
+                queuedDrivers = 0;
+                break;
+            case "RUNNING":
+                progress = query.totalDrivers == 0 ? 0 : Math.round((completedDrivers * 100) / query.totalDrivers);
+                if (query.isFullyBlocked) {
+                    summary = "BLOCKED";
+                    if (query.blockedReasons.length > 0) {
+                        summary += " (" + query.blockedReasons.join() + ")";
+                    }
+                }
+                else {
+                    summary = (progress == 0 ? summary : summary + " (" + progress + "%" + ")");
+                }
+                break;
+            case "FINISHED":
+                progress = 100;
+                runningDrivers = 0;
+                queuedDrivers = 0;
+        }
+        var progressBarStyle = {width: (progress == 0 ? 100 : progress) + "%", backgroundColor: getQueryStateColor(query.state, query.errorType, query.errorCode)};
+
+        var splitDetails = (
+            <div className="col-xs-12 tinystat-row">
+                <span className="tinystat" data-toggle="tooltip" data-placement="top" title="Completed splits">
+                    <span className="glyphicon glyphicon-ok" style={GLYPHICON_HIGHLIGHT}></span>&nbsp;&nbsp;
+                    { completedDrivers }
+                </span>
+                <span className="tinystat" data-toggle="tooltip" data-placement="top" title="Running splits">
+                    <span className="glyphicon glyphicon-play" style={GLYPHICON_HIGHLIGHT}></span>&nbsp;&nbsp;
+                    { runningDrivers }
+                </span>
+                <span className="tinystat" data-toggle="tooltip" data-placement="top" title="Queued splits">
+                    <span className="glyphicon glyphicon-pause" style={GLYPHICON_HIGHLIGHT}></span>&nbsp;&nbsp;
+                    { queuedDrivers }
```

Comment:
You're inconsistent about having or not having spaces before/after the braces. Make them consistent